arangod/RocksDBEngine/RocksDBIndex.cpp

```diff
@@ -274,6 +274,11 @@ Result RocksDBIndex::updateInternal(transaction::Methods* trx, RocksDBMethods* m
   // It is illegal to call this method on the primary index
   // RocksDBPrimaryIndex must override this method accordingly
   TRI_ASSERT(type() != TRI_IDX_TYPE_PRIMARY_INDEX);
+
+  // Indexing can only be disabled for non-unique indexes as they
+  // do not require an internal get after the removal
+  auto guard = DisableIndexingOnRocksDBTrxGuard(mthd, !_unique && trx->hasHint(transaction::Hints::Hint::FROM_AQL));
```

Comment:
mmmh. we disable indexing here (in RocksDBIndex.cpp), but we also disable it again in the nested `removeInternal` function. Is that sensible?